Sources/ScreenObject/ScreenObject.swift

```diff
@@ -93,15 +93,30 @@ open class ScreenObject {
             activityDescription = "Confirm whole screen \(self) is loaded"
         }
 
-        try XCTContext.runActivity(named: activityDescription) { (activity) in
-            try gettersToTest.forEach { getter in
-                let result = waitFor(
-                    element: getter(app),
-                    predicate: "isHittable == true",
-                    timeout: self.waitTimeout
-                )
-
-                guard result == .completed else { throw WaitForScreenError.timedOut }
+        var currentRetryCount = 0
+        while currentRetryCount < maxRetries {
+            currentRetryCount += 1
+            do {
+                try XCTContext.runActivity(named: activityDescription) { (activity) in
+                    try gettersToTest.forEach { getter in
+                        let result = waitFor(
+                            element: getter(app),
+                            predicate: "isHittable == true",
+                            timeout: self.waitTimeout
+                        )
+
+                        guard result == .completed else { throw WaitForScreenError.timedOut }
+                    }
+                }
+                break
+            } catch {
+                if currentRetryCount < maxRetries {
+                    // Wait 1 second before retrying
+                    sleep(1)
+                    continue
+                } else {
+                    throw WaitForScreenError.timedOut
+                }
```

Comment:
I'm not sure if this is the correct syntax, but we should be able to forward the error thrown from the `do try` part of the code, without having to replicate it.

```suggestion
            } catch error {
                if currentRetryCount < maxRetries {
                    // Wait 1 second before retrying
                    sleep(1)
                    continue
                } else {
                    throw error
                }
```